core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -315,19 +315,23 @@ class KafkaApis(val requestChannel: RequestChannel,
 
     // the callback for sending a produce response
     def sendResponseCallback(responseStatus: Map[TopicAndPartition, ProducerResponseStatus]) {
-      var errorInResponse = false
+
       val mergedResponseStatus = responseStatus ++ unauthorizedRequestInfo.mapValues(_ => ProducerResponseStatus(ErrorMapping.TopicAuthorizationCode, -1))
 
+      var errorInResponse = false
+
       mergedResponseStatus.foreach { case (topicAndPartition, status) =>
-        // we only print warnings for known errors here; if it is unknown, it will cause
-        // an error message in the replica manager
-        if (status.error != ErrorMapping.NoError && status.error != ErrorMapping.UnknownCode) {
-          debug("Produce request with correlation id %d from client %s on partition %s failed due to %s".format(
-            produceRequest.correlationId,
-            produceRequest.clientId,
-            topicAndPartition,
-            ErrorMapping.exceptionNameFor(status.error)))
+        if (status.error != ErrorMapping.NoError) {
           errorInResponse = true
+          // we only print warnings for known errors here; if it is unknown, it will cause
+          // an error message in the replica manager
+          if (status.error != ErrorMapping.UnknownCode) {
```

Comment:
Sounds good. Many other requests have similarly looking code, should those be changed as well or should we only change it here?